### PR TITLE
Wikidata tools v2

### PR DESCRIPTION
**Pull request by ImreSamu**
NEW version of shape file updating ( with wikidata names )
new version of  #245 , based on #249  ( #249 need for testing )

Actual status:  see end of the travis log : 
[Build Status](https://travis-ci.org/ImreSamu/natural-earth-vector)

added - new files  
```
./tools/wikidata/Makefile  
./tools/wikidata/README.md  
./tools/wikidata/update.sh  
./tools/wikidata/fetch_wikidata.py 
./tools/wikidata/write_wikidata.py
...
```

see more  ./tools/wikidata/README.md  

current status ( 2018May13) 
* https://gist.github.com/ImreSamu/a9d23c8f723cf135966eb1d10ce429a3

for example: 

### temp_shape/10m_cultural/ne_10m_populated_places.shp
 
shapefilename                             |  var                     |  value  | my comment |
------------------------------------------|--------------------------|------- | ------ |
10m_cultural/ne_10m_populated_places.shp  |  New_name                |  72655  | ( set a new name )
10m_cultural/ne_10m_populated_places.shp  |  Deleted_name            |  28     | ( old name is removed )
10m_cultural/ne_10m_populated_places.shp  |  Modified_name           |  266    | ( old name is modified )
10m_cultural/ne_10m_populated_places.shp  |  Empty_name              |  32658 | ( missing name )
10m_cultural/ne_10m_populated_places.shp  |  Same_name               |  44879  | ( name not changed )
10m_cultural/ne_10m_populated_places.shp  |  Wikidataid_redirected   |  3  | ( wikidataid changed, and updated in the shape file )
10m_cultural/ne_10m_populated_places.shp  |  Wikidataid_notfound     |  0  | ( has a wikidataid but got empty result from the SPARQL )
10m_cultural/ne_10m_populated_places.shp  |  Wikidataid_null         |  177 |  ( missing )
10m_cultural/ne_10m_populated_places.shp  |  Wikidataid_notnull      |  7166 | ( has a wikidata! )
10m_cultural/ne_10m_populated_places.shp  |  Wikidataid_badformated  |  0  | ( wikidataid not start with Q

### part of the ne_10m_populated_places.shp.changes_log.csv.md

wd_id      |  status    |  variable    |  value_old                              |  value_new
-----------|------------|--------------|-----------------------------------------|------------------------------------------------------------------
Q56064     |  NEWvalue  |  name_bn     |                                         |  কোলোনিয়া দেল সাক্রামেন্তো
Q56064     |  NEWvalue  |  name_el     |                                         |  Κολόνια ντελ Σακραμέντο
Q56064     |  NEWvalue  |  name_hu     |                                         |  Colonia del Sacramento
Q56064     |  NEWvalue  |  name_it     |                                         |  Colonia del Sacramento
Q56064     |  NEWvalue  |  name_ja     |                                         |  コロニア・デル・サクラメント
Q56064     |  NEWvalue  |  name_ko     |                                         |  콜로니아델사크라멘토
Q56064     |  NEWvalue  |  name_nl     |                                         |  Colonia del Sacramento
Q56064     |  NEWvalue  |  name_pl     |                                         |  Colonia del Sacramento
Q56064     |  MODvalue  |  name_pt     |  Colônia do Sacramento                  |  Colónia do Sacramento
Q56064     |  NEWvalue  |  name_sv     |                                         |  Colonia del Sacramento
Q56064     |  NEWvalue  |  name_tr     |                                         |  Colonia del Sacramento
Q56064     |  NEWvalue  |  name_vi     |                                         |  Colonia del Sacramento
Q598704    |  REDIRECT  |  wikidataid  |  Q598704                                |  Q22062855
Q1805374   |  REDIRECT  |  wikidataid  |  Q1805374                               |  Q22062457
Q2102493   |  REDIRECT  |  wikidataid  |  Q2102493                               |  Q22060164
Q130391    |  MODvalue  |  name_en     |  Tây Ninh                               |  Tay Ninh City
Q36354     |  MODvalue  |  name_en     |  Hà Tĩnh                                |  Hai Duong
Q34789     |  MODvalue  |  name_en     |  Vĩnh Long                              |  Vinh Long City
Q200441    |  MODvalue  |  name_en     |  Antioch                                |  Antioch on the Orontes
Q104400    |  MODvalue  |  name_en     |  Hamar                                  |  Hamar Municipality
Q206760    |  MODvalue  |  name_en     |  Kyrenia                                |  Keryneia
Q1026296   |  MODvalue  |  name_en     |  Jinotega City                          |  Municipality of Jinotega
Q2669506   |  MODvalue  |  name_en     |  Balancán                               |  Balancán Municipality
Q1196666   |  MODvalue  |  name_en     |  Lunan                                  |  Lunan District
Q1027910   |  MODvalue  |  name_en     |  Huangyan                               |  Huangyan District
Q145015    |  MODvalue  |  name_en     |  Zalantun                               |  Zhalantun
Q34761     |  MODvalue  |  name_en     |  Fularji                                |  Fularji District
Q649215    |  MODvalue  |  name_en     |  Baiquan                                |  Baiquan County
Q1065035   |  MODvalue  |  name_en     |  Sunshine Coast                         |  Sunshine Coast Region
Q2357934   |  MODvalue  |  name_en     |  Kerki                                  |  Atamyrat
Q33469     |  MODvalue  |  name_en     |  Ban Me Thuot                           |  Buôn Ma Thuột
Q14314     |  MODvalue  |  name_en     |  Santiago de Compostela                 |  Campillo del Río
Q200441    |  MODvalue  |  name_en     |  Antioch                                |  Antioch on the Orontes

ouch ...

the "Santiago de Compostela" ->  "Campillo del Río" - probably a fake rename ( anonym user (ip addr)  via mobil edit )   https://www.wikidata.org/w/index.php?title=Q14314&action=history
screenshot from 2018-05-13 16-22-48

I have fixed about a dozen fake renames  ... so we need a manual check  :(

### CODE REVIEW

**Comment by nvkelso** [comment]
Hi! I think my WIP branch is "done" pending final QA pass and integration of your work here in this branch! All the admin-0 and admin-1 features have been updated over your last merge, and now includes 99% wikidata coverage for admin-0 (for countries like Peru).

**Comment by ImreSamu** [comment]
Hi @nvkelso ,

TLDR:  most of the code is OK on my side;    please test, if it working on your computer ...


I have finished the most of the coding, ( ~94% )
* the open part is the name cleaning ..  ( Removing: "city" ,"lake" words from the names )

As I see - it is running OK -  in the travis CI - for   `ne_10m_*` tables
 * see logs:  https://travis-ci.org/ImreSamu/natural-earth-vector

Current readme :
*  https://github.com/ImreSamu/natural-earth-vector/blob/5f8521b270b355e48c0b8e4cd455ad6d2ea5c300/tools/wikidata/README.md


now - it is working with:
* uppercase variable names[WIKIDATAID,NAME_AR,NAME_BN,NAME_DE,NAME_EN,NAME_ES, ... ]
* lowercase variable names[wikidataid, name_ar, name_bn, name_de, name_en, name_es, ... ]

my metadata in the `update.sh`

```bash
#  lettercase = uppercase variable names [WIKIDATAID, NAME_AR, NAME_BN, NAME_DE, NAME_EN, NAME_ES, ... ]
update_names uppercase 10m_cultural ne_10m_admin_0_countries_lakes.shp
update_names uppercase 10m_cultural ne_10m_admin_0_countries.shp
update_names uppercase 10m_cultural ne_10m_admin_0_disputed_areas.shp
update_names uppercase 10m_cultural ne_10m_admin_0_map_subunits.shp
update_names uppercase 10m_cultural ne_10m_admin_0_map_units.shp
update_names uppercase 10m_cultural ne_10m_admin_0_sovereignty.shp


#  lettercase = lowercase variable names [wikidataid, name_ar, name_bn, name_de, name_en, name_es ]
update_names lowercase 10m_cultural ne_10m_admin_1_states_provinces_lakes.shp
update_names lowercase 10m_cultural ne_10m_admin_1_states_provinces.shp
update_names lowercase 10m_cultural ne_10m_airports.shp
update_names lowercase 10m_cultural ne_10m_populated_places.shp
update_names lowercase 10m_physical ne_10m_geographic_lines.shp
update_names lowercase 10m_physical ne_10m_geography_marine_polys.shp
update_names lowercase 10m_physical ne_10m_geography_regions_elevation_points.shp
update_names lowercase 10m_physical ne_10m_geography_regions_points.shp
update_names lowercase 10m_physical ne_10m_geography_regions_polys.shp
update_names lowercase 10m_physical ne_10m_lakes_europe.shp
update_names lowercase 10m_physical ne_10m_lakes_historic.shp
update_names lowercase 10m_physical ne_10m_lakes_north_america.shp
update_names lowercase 10m_physical ne_10m_lakes.shp
update_names lowercase 10m_physical ne_10m_playas.shp
update_names lowercase 10m_physical ne_10m_rivers_europe.shp
update_names lowercase 10m_physical ne_10m_rivers_lake_centerlines_scale_rank.shp
update_names lowercase 10m_physical ne_10m_rivers_lake_centerlines.shp
update_names lowercase 10m_physical ne_10m_rivers_lake_centerlines.shp
update_names lowercase 10m_physical ne_10m_rivers_north_america.shp

#WARNING:Fiona:CPLE_AppDefined in b'Value -4.75267000000000017 of field longitude of feature 4645 not successfully written. Possibly due to too larger number with respect to field width'
#WARNING:Fiona:CPLE_AppDefined in b'Value 10.2509999999999994 of field latitude of feature 4646 not successfully written. Possibly due to too larger number with respect to field width'
#WARNING:Fiona:CPLE_AppDefined in b'Value -3.34011000000000013 of field longitude of feature 4646 not successfully written. Possibly due to too larger number with respect to field width'
update_names lowercase 10m_cultural ne_10m_admin_1_label_points_details.shp
```


### Example ne_10m_admin_0_countries  ( The modified  country names )
```bash
$ cat temp_shape/10m_cultural/ne_10m_admin_0_countries.shp.changes_log.csv.md | grep MOD
Q184      |  MODvalue  |  NAME_RU   |  Республика Беларусь         |  Белоруссия
Q184      |  MODvalue  |  NAME_ZH   |  白俄罗斯                        |  白罗斯
Q881      |  MODvalue  |  NAME_HU   |  Vietnam                     |  Vietnám
Q1246     |  MODvalue  |  NAME_EL   |  Κόσοβο                      |  Κοσσυφοπέδιο
Q977      |  MODvalue  |  NAME_PT   |  Jibuti                      |  Djibouti
Q986      |  MODvalue  |  NAME_AR   |  إريتريا                     |  إرتريا
Q1050     |  MODvalue  |  NAME_ES   |  Suazilandia                 |  eSwatini
Q1050     |  MODvalue  |  NAME_PT   |  Suazilândia                 |  eSwatini
Q1050     |  MODvalue  |  NAME_SV   |  Swaziland                   |  eSwatini
Q836      |  MODvalue  |  NAME_FR   |  Birmanie                    |  Myanmar
Q736      |  MODvalue  |  NAME_EL   |  Ισημερινός                  |  Εκουαδόρ
Q217      |  MODvalue  |  NAME_DE   |  Moldawien                   |  Republik Moldau
Q30       |  MODvalue  |  NAME_AR   |  الولايات المتحدة الأمريكية  |  الولايات المتحدة
Q223      |  MODvalue  |  NAME_AR   |  غرينلاند                    |  جرينلاند
Q865      |  MODvalue  |  NAME_DE   |  Republik China              |  Taiwan
Q865      |  MODvalue  |  NAME_PT   |  República da China          |  Taiwan
Q1042     |  MODvalue  |  NAME_RU   |  Сейшелы                     |  Сейшельские Острова
Q760      |  MODvalue  |  NAME_IT   |  Santa Lucia                 |  Saint Lucia
Q5785     |  MODvalue  |  NAME_RU   |  Каймановы острова           |  острова Кайман
Q25230    |  MODvalue  |  NAME_ES   |  Bailía del Guernsey         |  Guernsey
Q685      |  MODvalue  |  NAME_FR   |  Salomon                     |  Îles Salomon
Q628716   |  MODvalue  |  NAME_SV   |  Panatag                     |  Scarboroughrevet

$ cat temp_shape/10m_cultural/ne_10m_admin_0_countries.shp.summary_log.csv.md
shapefilename                              |  var                     |  value
-------------------------------------------|--------------------------|-------
10m_cultural/ne_10m_admin_0_countries.shp  |  New_name                |  444
10m_cultural/ne_10m_admin_0_countries.shp  |  Deleted_name            |  0
10m_cultural/ne_10m_admin_0_countries.shp  |  Modified_name           |  22
10m_cultural/ne_10m_admin_0_countries.shp  |  Empty_name              |  23
10m_cultural/ne_10m_admin_0_countries.shp  |  Same_name               |  4866
10m_cultural/ne_10m_admin_0_countries.shp  |  Wikidataid_redirected   |  0
10m_cultural/ne_10m_admin_0_countries.shp  |  Wikidataid_notfound     |  0
10m_cultural/ne_10m_admin_0_countries.shp  |  Wikidataid_null         |  0
10m_cultural/ne_10m_admin_0_countries.shp  |  Wikidataid_notnull      |  255
10m_cultural/ne_10m_admin_0_countries.shp  |  Wikidataid_badformated  |  0

```

**Comment by nvkelso** [comment]
I checked out the new code on my Mac running 10.13.3 (17D47) but ran into problem (after installing all dependencies):

```
$ time ./tools/wikidata/update.sh fetch
-bash: ./tools/wikidata/update.sh: /bin/bash^M: bad interpreter: No such file or directory

real	0m0.001s
user	0m0.000s
sys	0m0.001s
```

Any ideas?

**Comment by nvkelso** [comment]
> the open part is the name cleaning .. ( Removing: "city" ,"lake" words from the names )

Another thing I've noticed with county labels in particular is often the longest name translation (the formal name) is preferred when really it's the shorter one we want.

**Comment by nvkelso** [comment]
And for fun the QGIS3 doc has been updated to show all the alternate languages with a project variable!

Seems like sometimes an Indic script is being set for Hungarian, though?

screen shot 2018-05-15 at 23 43 46
screen shot 2018-05-15 at 23 43 22
screen shot 2018-05-15 at 23 42 56
screen shot 2018-05-15 at 23 42 22

screen shot 2018-05-15 at 23 45 35

**Comment by nvkelso** [comment]
I suspect there's something fishy with my Python 3 install on this MacBook... Will debug on another machine...

**Comment by nvkelso** [comment]
The solution here does resolve around bad line encoding.

`$ ./tools/wikidata/update.sh fetch`

Which returns:

`-bash: ./tools/wikidata/update.sh: /bin/bash^M: bad interpreter: No such file or directory`

Following the instructions here:

- https://stackoverflow.com/questions/2920416/configure-bin-shm-bad-interpreter/11019077

I replace the line encodings and then the script runs!

But but but...

```
- Start fetching Natural-Earth wikidata labels via SPARQL query - 
fetch:  wd:Q833 ...  wd:Q760
ERRwikidata: other error. Retrying in 3 seconds.
ERRwikidata: other error. Retrying in 3 seconds.
ERRwikidata: other error. Retrying in 3 seconds.
ERRwikidata: other error. Retrying in 3 seconds.
ERRwikidata: other error. Retrying in 3 seconds.
ERRwikidata: other error. Retrying in 3 seconds.
ERRwikidata: other error. Retrying in 3 seconds.
ERRwikidata: other error. Retrying in 3 seconds.
Wikidata request failed ; system stopped! 
```

Which isn't so helpful :\ Any ideas on why this is failing?

**Comment by nvkelso** [comment]
Since there are lot of suggestions there, for the record I used **vim**:

To fix, open your script with vi or vim:

> vi tools/wikidata/update.sh

and enter in vi command mode (key Esc or just `:` key), then type this:

> :set fileformat=unix

Finally save it:

> :x!

**Comment by nvkelso** [comment]
After fixing the line encodings in the shell script again with Vim, I get different / more detailed errors:

```
- Start fetching Natural-Earth wikidata labels via SPARQL query - 
fetch:  wd:Q833 ...  wd:Q760
ERRwikidata: other error. Retrying in 3 seconds.
error: <class 'urllib.error.URLError'> 
ERRwikidata: other error. Retrying in 3 seconds.
error: <class 'urllib.error.URLError'> 
ERRwikidata: other error. Retrying in 3 seconds.
error: <class 'urllib.error.URLError'> 
ERRwikidata: other error. Retrying in 3 seconds.
error: <class 'urllib.error.URLError'> 
ERRwikidata: other error. Retrying in 3 seconds.
error: <class 'urllib.error.URLError'> 
ERRwikidata: other error. Retrying in 3 seconds.
error: <class 'urllib.error.URLError'> 
ERRwikidata: other error. Retrying in 3 seconds.
error: <class 'urllib.error.URLError'> 
ERRwikidata: other error. Retrying in 3 seconds.
error: <class 'urllib.error.URLError'> 
Wikidata request failed ; system stopped! 

real	0m26.276s
user	0m0.408s
sys	0m0.406s
```

**Comment by nvkelso** [comment]
Maybe for now just send me your output and I'll figure it out locally so we can push the v4.1 changes out sooner while this computer funk is being debugged?

**Comment by nvkelso** [comment]
Running:

> $ python3 ./tools/wikidata/platform_debug_info.py

Result:

```
system       : Darwin
release      : 17.4.0
version      : Darwin Kernel Version 17.4.0: Sun Dec 17 09:19:54 PST 2017; root:xnu-4570.41.2~1/RELEASE_X86_64
machine      : x86_64
processor    : i386
Python       : 3.6.5
Fiona        : 1.7.11.post2
SPARQLWrapper: 1.8.1
requests     : 2.18.4
```

**Comment by nvkelso** [comment]
Looks like I'm on a newer major Darwin (macOS) version.
Same Python.
Same requests.

```
Fiona        : 1.7.11 (I'm on 1.7.11.post2)
SPARQLWrapper: 1.8.0 (I'm on 1.8.1)
```

**Comment by nvkelso** [comment]
Yes, Docker works on this machine.

**Comment by ImreSamu** [comment]
thanx,   ( probably a proxy problem? ...  strange ... )

>  Yes, Docker works on this machine.

ok ... 
 I try  to create a docker images ...

**Comment by nvkelso** [comment]
Some more debug... after killing the fetch script it reports:

> ssl.SSLError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:833)

**Comment by ImreSamu** [comment]
docker version - is ready for testing ..
```
git clone --depth=1 --branch=wikidata_tools_v2 https://github.com/ImreSamu/natural-earth-vector.git ImreSamu/natural-earth-vector
cd ImreSamu/natural-earth-vector
# build docker image
docker build -t ne_py3wikidata .
# run - inside docker image 
docker run -it -v $(pwd):/ne   ne_py3wikidata  bash run_all.sh
```

> [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:833)

certification?
according to http://docs.python-requests.org/en/master/user/advanced/#ca-certificates  maybe too old.
try?    ` pip3 install -U certifi `

**Comment by nvkelso** [comment]
OMG, this solved it for me:

https://github.com/Shopify/shopify_python_api/issues/201#issuecomment-345880213

Namely:

> I have the same issue on macOS with Python 3.6.0. This solution—detailed at https://stackoverflow.com/a/42098127/592820—resolved the problem for me:

Run `/Applications/Python\ 3.6/Install\ Certificates.command` from a terminal.

Which logs:

```
 -- pip install --upgrade certifi
Requirement already up-to-date: certifi in /Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages (2018.4.16)
 -- removing any existing file or link
 -- creating symlink to certifi certificate bundle
 -- setting permissions
 -- update complete
```

And then (after re-applying the linux line breaks), the scripts work (without even getting into Docker!).

**Comment by nvkelso** [comment]
In populated places, these and similar shouldn't have ` City` stripped:

> Q169948    |  MODvalue  |  name_en     |  Yuba City                              |  Yuba
> Q166866    |  MODvalue  |  name_en     |  Crescent City                          |  Crescent

**Comment by ImreSamu** [comment]
> the scripts work (without even getting into Docker!).

Good news :+1: 

> after re-applying the linux line breaks)

as I see - there is a git  _"global settings for line endings"_
 *  https://help.github.com/articles/dealing-with-line-endings/#platform-mac

maybe a `.gitattributes`  file for the project -   is solving this.
 * https://help.github.com/articles/dealing-with-line-endings/#per-repository-settings
 * https://stackoverflow.com/questions/47468094/line-endings-with-git

File: tools/wikidata/README.md [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+* 10m_physical/ne_10m_geography_regions_polys.shp
+* 10m_physical/ne_10m_lakes_europe.shp
+* 10m_physical/ne_10m_lakes_historic.shp
+* 10m_physical/ne_10m_lakes_north_america.shp
+* 10m_physical/ne_10m_lakes.shp
+* 10m_physical/ne_10m_playas.shp
+* 10m_physical/ne_10m_rivers_europe.shp
+* 10m_physical/ne_10m_rivers_lake_centerlines_scale_rank.shp
+* 10m_physical/ne_10m_rivers_lake_centerlines.shp
+* 10m_physical/ne_10m_rivers_lake_centerlines.shp
```
Comment (nvkelso) [inline tools/wikidata/README.md]: Duplicate line

File: tools/wikidata/update.sh [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+update_names lowercase    .              x_tempshape     10m_physical   ne_10m_geography_regions_polys.shp
+update_names lowercase    .              x_tempshape     10m_physical   ne_10m_lakes_europe.shp
+update_names lowercase    .              x_tempshape     10m_physical   ne_10m_lakes_historic.shp
+update_names lowercase    .              x_tempshape     10m_physical   ne_10m_lakes_north_america.shp
+update_names lowercase    .              x_tempshape     10m_physical   ne_10m_lakes.shp
+update_names lowercase    .              x_tempshape     10m_physical   ne_10m_playas.shp
+update_names lowercase    .              x_tempshape     10m_physical   ne_10m_rivers_europe.shp
+update_names lowercase    .              x_tempshape     10m_physical   ne_10m_rivers_lake_centerlines_scale_rank.shp
+update_names lowercase    .              x_tempshape     10m_physical   ne_10m_rivers_lake_centerlines.shp
+update_names lowercase    .              x_tempshape     10m_physical   ne_10m_rivers_lake_centerlines.shp
```
Comment (nvkelso) [inline tools/wikidata/update.sh]: Duplicate line

**Comment by nvkelso** [comment]
The following will also need the name lookup treatments (they are either built using MapShaper with a data join or have their own special data because of their scale set):

- [ ] ne_50m_admin_0_sovereignty
- [ ] ne_50m_admin_0_countries
- [ ] ne_50m_admin_0_countries_lakes
- [ ] ne_50m_admin_0_map_units
- [ ] ne_50m_admin_0_map_subunits
- [ ] ne_50m_admin_0_tiny_countries
- [ ] ne_50m_admin_0_breakaway_disputed_areas
- [ ] ne_50m_admin_0_breakaway_disputed_areas_scale_rank
- [ ] ne_50m_admin_1_states_provinces
- [ ] ne_50m_admin_1_states_provinces_lakes
- [ ] ne_50m_lakes
- [ ] ne_50m_lakes_historic
- [ ] ne_50m_playas
- [ ] ne_50m_rivers_lake_centerlines
- [ ] ne_50m_rivers_lake_centerlines_scale_rank
- [ ] ne_110m_admin_0_sovereignty
- [ ] ne_110m_admin_0_countries
- [ ] ne_110m_admin_0_countries_lakes
- [ ] ne_110m_admin_0_map_units
- [ ] ne_110m_admin_1_states_provinces
- [ ] ne_110m_admin_1_states_provinces_lakes
- [ ] ne_110m_lakes
- [ ] ne_110m_rivers_lake_centerlines

**Comment by nvkelso** [comment]
This one is odd (the emoji flag):

```
/ne_10m_admin_0_sovereignty.shp) 
---------------------
Q733      |  MODvalue  |  NAME_EN   |  Paraguay                    |  Brazil 🇧🇷
```

**Comment by ImreSamu** [comment]
> This one is odd (the emoji flag):

:(

this is reverted now ( Revision history of "Paraguay" (Q733) )  
* it was valid:    03:38, 20 May 2018  -  05:23, 20 May 2018‎ 
https://www.wikidata.org/w/index.php?title=Q733&action=history

so re-running the "fetch" process is solving this issue ( but maybe someone creating other problems...  )  

this is one of my top priority   -  detecting this type of edits (  made by not registered users )

**Comment by nvkelso** [comment]
Probably after a few rounds of editing the country names we should **_not_** allow modifications on them (only writes for something that was null)?

**Comment by nvkelso** [comment]
I'm pretty happy with where this PR is :)

Once these 4 issues are taken care of I think we can merge it and then do any cleanup in subsequent PR:

- https://github.com/nvkelso/natural-earth-vector/pull/258#pullrequestreview-121637199 (dup line)
- https://github.com/nvkelso/natural-earth-vector/pull/258#pullrequestreview-121637204 (dup line)
- https://github.com/nvkelso/natural-earth-vector/pull/258#discussion_r189452467 (extra themes)
- https://github.com/nvkelso/natural-earth-vector/pull/258#issuecomment-390460592 (.gitattributes)

I've also been experimenting on formatting the GeoJSON output to be the same property indented formatting (but geometry on single line) as Who's On First – that would make the diffs easy to read as to which properties are changing run to run.

Timing wise I'd like to get this all sorted this weekend, worst case Tuesday night.

**Comment by ImreSamu** [comment]
still working ...

**Comment by ImreSamu** [comment]
ready for test.

#### 1.)  "4 issues"  - I have added ..
#### 2.)  big refactoring -  improved tools  - for processing individual shape files , 

so it can be run step by step ..  ( see [`./run_all.sh`](https://github.com/ImreSamu/natural-earth-vector/blob/bf1747b351ce800bf2431edade6f097fff6c3a8d/run_all.sh)   )

example for : `10m_cultural    ne_10m_admin_0_countries`
```bash
# fetch
./tools/wikidata/update.sh  fetch  uppercase   10m_cultural  ne_10m_admin_0_countries
# write
./tools/wikidata/update.sh  write  uppercase   10m_cultural  ne_10m_admin_0_countries
# fetch + write
./tools/wikidata/update.sh  fetch_write uppercase   10m_cultural  ne_10m_admin_0_countries
# copy ( to the original place )
./tools/wikidata/update.sh  copy  uppercase   10m_cultural  ne_10m_admin_0_countries
# all ( fetch + write + copy )
./tools/wikidata/update.sh  all  uppercase   10m_cultural  ne_10m_admin_0_countries
```

#### 3.)  "extra themes" - 2 files has no `wikidatid`  ( the others - OK )
     * ne_50m_admin_0_breakaway_disputed_areas             
     * ne_50m_admin_0_breakaway_disputed_areas_scale_rank  # KeyError: 'WIKIDATAID'

#### 4.) and other minor improvements.  ....


#### 5.) .travis test ok  https://travis-ci.org/ImreSamu/natural-earth-vector 
 
now it is more complex because of the .travis timeout workaround  

now:
   * osx   :  processing  110m_*   +  `make clean all`     ( minimal fast test )
   * linux :  processing  10m_*  +  50m_*  +  110m_*  +   `make clean all`   ( run everything )

**Comment by nvkelso** [comment]
Also fixed Persian Gulf: https://www.wikidata.org/wiki/Q34675.

**Comment by nvkelso** [comment]
I fixed up a few things and merged this, huzzah!

- 50m disputed admin-0 funk
- more targeted copying of result files out of tmp over originals (don't copy csv or md)
- reordered a few Wikidata lookups so they match other NE orders (no-op)
- comment out admin-1 label points for now – doesn't matter since the output files from that are also processed
- Few other housekeeping things for v4.1 CHANGELOG / Makefile that got swept up here.

Thank you so much @ImreSamu for your big lift here, amazing work linking up Natural Earth with Wikidata and opening the projects to a wider audience! :)

**Comment by ImreSamu** [comment]
Thanks!!!! 
wikidata_stamp_small_v2

[wikidata_stamp.png](https://commons.wikimedia.org/wiki/File:Wikidata_stamp.png)

We need an update here:  
* https://en.wikipedia.org/wiki/Natural_Earth
* https://www.wikidata.org/wiki/Q6980476  ( Natural Earth (Q6980476) )

And please turn on the `.travis` checking :  
* https://travis-ci.org/nvkelso/natural-earth-vector

**Comment by nvkelso** [comment]
This has been released on the public download site: https://www.naturalearthdata.com/blog/miscellaneous/natural-earth-v4-1-0-release-notes/
